### ISSUE TITLE
mrr offline rigs

### ISSUE THREAD

**Issue by ciumete** [RainbowMiner/RainbowMiner#2371]
[2023-05-25 09:39:43] WARNING: MiningRigRentals: Recovery mode is enabled! Please disable it after all your rigs have been recovered.
[2023-05-25 09:39:44] INFO: MiningRigRentals: Recovery mode enabled - 82 orphaned rigs found
[2023-05-25 09:39:45] INFO: MiningRigRentals: Updating algorithms config data (forced)
[2023-05-25 09:39:45] INFO: MiningRigRentals: Start create rigs on rig3
[2023-05-25 09:39:46] INFO: MiningRigRentals: Start update rigs on rig3
[2023-05-25 09:39:46] INFO: MiningRigRentals: Update rig #201436 Autolykos2 [rig3]: hash=240mh, minimum=0.43726086008/th/day, minhours=3, ndevices=1, device_ram=6.921, modifier=, region=eu, extensions=True
[2023-05-25 09:39:48] INFO: MiningRigRentals loaded in 6.782s 
.
.
.
[2023-05-25 11:00:02] WARNING: MiningRigRentals: Recovery mode is enabled! Please disable it after all your rigs have been recovered.
[2023-05-25 11:00:03] INFO: MiningRigRentals: Recovery mode enabled - 82 orphaned rigs found
[2023-05-25 11:00:04] INFO: MiningRigRentals: Updating algorithms config data (forced)
[2023-05-25 11:00:05] INFO: MiningRigRentals: Start create rigs on rig3
[2023-05-25 11:00:05] INFO: MiningRigRentals: Start update rigs on rig3
[2023-05-25 11:00:07] INFO: MiningRigRentals loaded in 5.828s

**Comment by sfalas01** [comment]
Yep, something in the latest update broke MRR completely for me.

**Comment by sfalas01** [comment]
I have no idea. The only way I managed to solve it was to reinstall from an older version and create the rigs again. After updating, it works. But if you try to recreate rigs from the latest version, it stops recreating them. Maybe something changed in the API of MRR, I have not isolated the problem yet

**Comment by sfalas01** [comment]
I know, what to do

**Comment by ciumete** [comment]
Nobody confirm if is a software issue or is my configuration issue !!!

**Comment by RainbowMiner** [comment]
Please try to create a new API key at Miningrigrentals and replace the old keys in pools.config.txt. MRR uses a special counter based security feature and for various reasons in some rare edge cases, the counter desyncs.

**Comment by ciumete** [comment]
> Please try to create a new API key at Miningrigrentals and replace the old keys in pools.config.txt. MRR uses a special counter based security feature and for various reasons in some rare edge cases, the counter desyncs.

i created 2 more pairs api keys, same result with noew keys, all rigs are offline !!!

[2023-06-18 20:35:31] INFO: MiningRigRentals: Start create rigs on rig3
[2023-06-18 20:35:31] INFO: MiningRigRentals: Start update rigs on rig3
[2023-06-18 20:35:32] INFO: MiningRigRentals: Update rig #177094 CryptoNightHeavyXhv [rig3]: hash=2.7kh, minimum=0.02040730388/mh/day, minhours=5, ndevices=1, device_ram=6.921, modifier=, region=eu, extensions=True
[2023-06-18 20:35:32] INFO: MiningRigRentals: Update rig #175601 CryptoNightLiteUpx2 [rig3]: hash=86.3kh, minimum=0.783244004058/gh/day, minhours=4, ndevices=1, device_ram=6.921, modifier=, region=eu, extensions=True
[2023-06-18 20:35:32] INFO: MiningRigRentals: Update rig #208056 RandomArq [rig3]: hash=14.2kh, minimum=0.004946173503/mh/day, minhours=4, ndevices=1, device_ram=6.921, modifier=, region=eu, extensions=True
[2023-06-18 20:35:33] INFO: MiningRigRentals loaded in 5.136s

**Comment by RainbowMiner** [comment]
Oh, I see! So the rigs are being updated successfully.
Could you please create a Debug file and upload.it here? Click "Debug file" on the left hand side on localhost:4000 to create that file.
I will then check your settings.

**Comment by ciumete** [comment]
[debug_2023-06-18.zip](https://github.com/RainbowMiner/RainbowMiner/files/11782302/debug_2023-06-18.zip)

**Comment by ciumete** [comment]
> Thanks, I will check this asap.

you find the problem ? it worked for years with this configuration !

**Comment by ciumete** [comment]
i made some test with only one rig who apear online and have same configuration with others.
the rig0 working ok with mrr, shows online and calculate different the price from others rigs, but i compared config and is identical !!!

rig0 online rig on mrr:
[2023-07-02 22:52:51] INFO: MiningRigRentals: Start update rigs on rig0
[2023-07-02 22:52:51] INFO: MiningRigRentals: Update rig #199474 VerusHash [rig0]: hash=3.5mh, minimum=0.029493981139/gh/day, minhours=3, ndevices=1, device_ram=6.922, modifier=, region=eu, extensions=True

rig3 offline rig on mrrr
[2023-07-02 22:38:22] INFO: MiningRigRentals: Start update rigs on rig3
[2023-07-02 22:38:22] INFO: MiningRigRentals: Update rig #285323 VerusHash [rig3]: hash=23.1mh, minimum=0.003072242081/gh/day, minhours=4, ndevices=1, device_ram=6.921, modifier=, region=eu, extensions=True

so minimum is ten times smaller in rig3 compared with rig0 !!!
minhours in config is on both rigs 3 hours but on rig3 updated mrr with 4 hours!

**Comment by ciumete** [comment]
i reinstalled software, i created a new account on mrr site, same issue !!! nothing change!!! one old rig is online, others are offline !!! very strange issue !!!

**Comment by ciumete** [comment]
I think i found the problem. Is about one of mrr servers who is filtered on route. So is an internet connection problem.
